### PR TITLE
chore: export tool minor refactor

### PR DESCRIPTION
**Pull request by shuiyisong**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This pr is a cr issue fix to #2610 

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/cmd/src/cli/export.rs [hunk 1]
```diff
@@ -328,20 +332,25 @@ impl Export {
...
-
-                        format!(
-                            "copy {} from '{}' with (format='parquet');\n",
-                            filename.replace(".parquet", ""),
-                            file.path().to_str().unwrap()
-                        )
-                    })
-                    .collect::<Vec<_>>()
-                    .join("");
+                let copy_from_sql =
```
[the remark this answers is not in the document]
Comment (shuiyisong) [reply src/cmd/src/cli/export.rs:335]: The idea is try to reduce the number of IO operation to avoid random IO failure.

**Comment by killme2008** [comment]
@shuiyisong PLTAL

**Comment by shuiyisong** [comment]
@evenyag @waynexia PTAL 😀

File: src/cmd/src/cli/export.rs [hunk 2]
```diff
@@ -70,6 +73,10 @@ pub struct ExportCommand {
     /// Things to export
     #[clap(long, short = 't', value_enum)]
     target: ExportTarget,
+
+    /// basic authentication for connecting to the server
+    #[clap(long)]
+    auth_basic: Option<String>,
```
Comment (waynexia) [inline src/cmd/src/cli/export.rs:79]: maybe move this to an upper-level command option in the future, if it's an common requirement

**Review by waynexia — APPROVED** [review]
LGTM

File: src/cmd/src/cli/export.rs [hunk 3]
```diff
@@ -328,23 +348,26 @@ impl Export {
...
-                    })
-                    .collect::<Vec<_>>()
-                    .join("");
-                file.write_all(copy_from_sql.as_bytes())
+                        .as_bytes(),
+                    )
                     .await
                     .context(FileIoSnafu)?;
+                    file.flush().await.context(FileIoSnafu)?;
+                }
```
Comment (evenyag) [inline src/cmd/src/cli/export.rs:370]: We can move flush outside of the for loop.
```suggestion
                }
                file.flush().await.context(FileIoSnafu)?;
```

File: src/cmd/src/cli/export.rs [hunk 4]
```diff
@@ -328,23 +348,26 @@ impl Export {
                     Path::new(&self.output_dir).join(format!("{catalog}-{schema}_copy_from.sql"));
                 let mut file = File::create(copy_from_file).await.context(FileIoSnafu)?;
```
Comment (evenyag) [inline src/cmd/src/cli/export.rs:349]: nitpick(non-blocking): We can use a BufWriter to reduce IO operations.

**Review by evenyag — APPROVED** [review]
Rest LGTM.

**Pull request merged**